### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by jmehic**
**What's in this PR?**

This PR updates the test infrastructure to use the latest version of docker compose files in order to take advantage of some new compose features.  In particular it replaces the deprecated link logic with the new network functionality.  Additionally the test scripts were cleaned up and simplified, some hard-coded test resource paths were removed, and a bit of general re-organisation was done.

### CODE REVIEW

**Comment by unusedPhD** [comment]
You should remove the vendor folder and make sure it's part of the gitignore file.  No need to import that into the repo.

**Comment by jmehic** [comment]
can't reply directly but disagreed @unusedPhD . i specifically removed the vendor line from the gitignore so the folder would be committed. it's the best way to ensure reproducible builds and allow people to simply clone and run the project right away, without having to worry about dependency management (i.e. making sure they get each dependency at the right revision).
see https://github.com/golang/dep/blob/master/docs/FAQ.md#should-i-commit-my-vendor-directory

**Comment by unusedPhD** [comment]
Appears you would be inline with the authors of godep as well: https://github.com/tools/godep/blob/master/FAQ.md

File: db/test.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python2
```
Comment (amadeovezz) [inline db/test.py:1]: non-blocking for this PR, but at some point we might want to upgrade to python 3 🐍

**Comment by amadeovezz** [comment]
awesome work @jmehic, this looks good to me!

**Pull request merged**